.github/workflows/deploy.yml

```diff
@@ -0,0 +1,39 @@
+name: Deploy to GitHub Pages
+
+on:
+  # Triggers the workflow on push or pull request events but
+  # only for the default branch.
+  push:
+    branches:
+      - master
+    paths:
+      - '.github/workflows/deploy.yml'
+      - 'book.toml'
+      - 'guide/**'
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  fetch:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v4
+      - name: Install Rust packages
+        run: |
+          curl --proto '=https' --tlsv1.2 -sSf https://rossmacarthur.github.io/install/crate.sh | bash -s -- --repo "rust-lang/mdBook" --bin mdbook --to ~/.cargo/bin
+      - name: Install gh-pages
+        run: |
+          npm install gh-pages@"~6.1.1"
+      - name: Build book
+        run: |
+            mdbook --version
+            mdbook build
+      - name: Set Git config
+        run: |
+          git config --global user.email "actions@users.noreply.github.com"
+          git config --global user.name 'Automation'
+          git remote set-url origin https://x-access-token:${{ secrets.GITHUB_TOKEN }}@github.com/$GITHUB_REPOSITORY
```

Comment:
Is this last line necessary? If we add a top-level config
```yaml
permissions:
  contents: write
```
would that not have the same effect, without requiring a `GITHUB_TOKEN` getting passed around?

Reply:
Similarly, not sure this was available at the time. I'll test with the permissions bit.

Reply:
It doesn't work, I end up with an error
https://github.com/tschaub/gh-pages/issues/384

I'm guessing that I could create a PAT and use that, but I'd prefer avoiding it since this method works fine.